python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -131,41 +133,6 @@ def setParams(self, inputCol=None, inputImageNodeName=None, outputCol=None,
         kwargs = self._input_kwargs
         return self._set(**kwargs)
 
-    def fit(self, dataset, params=None):
-        """
-        Fits a model to the input dataset with optional parameters.
-
-        .. warning:: This returns the byte serialized HDF5 file for each model to the driver.
-                     If the model file is large, the driver might go out-of-memory.
-                     As we cannot assume the existence of a sufficiently large (and writable)
-                     file system, users are advised to not train too many models in a single
-                     Spark job.
-
-        :param dataset: input dataset, which is an instance of :py:class:`pyspark.sql.DataFrame`.
-                        The column `inputCol` should be of type `sparkdl.image.imageIO.imgSchema`.
-        :param params: An optional param map that overrides embedded params. If a list/tuple of
-                       param maps is given, this calls fit on each param map and returns a list of
-                       models.
-        :return: fitted model(s). If params includes a list of param maps, the order of these
-                  models matches the order of the param maps.
-        """
-        self._validateParams()
-        if params is None:
-            paramMaps = [dict()]
-        elif isinstance(params, (list, tuple)):
-            if len(params) == 0:
-                paramMaps = [dict()]
-            else:
-                self._validateFitParams(params)
-                paramMaps = params
-        elif isinstance(params, dict):
-            self._validateFitParams(params)
-            paramMaps = [params]
-        else:
-            raise ValueError("Params must be either a param map or a list/tuple of param maps, "
```

Comment:
Just to confirm: looks like in this PR we make a few breaking API changes:
1) We now require `KerasFitParams` to be a single param map dictionary & no longer support passing a list/tuple of param maps. 
2) `fit()` no longer accepts a `params` argument. 

For 1), I'm not sure if we need to support setting `KerasFitParams` to a list/tuple of param maps. IMO the question is whether using `KerasImageFileEstimator` without cross-validation is a legitimate use case (e.g. for hyperparameter tuning without a validation set). If it is legitimate, we should support passing a list/tuple (preferably just one of the two to simplify things for us, since we have an opportunity to break APIs with the DLP 1.0 release). If not, we can break the old behavior & should probably document that KIFE is intended to be used with CrossValidator. Curious to hear what @sueann & @tomasatdatabricks think here.

For 2) I'm fine with making this change - maybe for now we can keep `def fit(self, dataset, params=None)` and raise an exception describing the behavior change & how to deal with it if the user specifies `params`. We'll miss the edge case (i.e. not raise an exception) where users explicitly pass params=None, but hopefully almost nobody is doing that.

Let me know if I missed something!

Reply:
1. [KerasFitParams] is a dict that contains hyperparameters that Keras needs (these are independent of MLlib `Param`s)
2. [estimator:fit] calls `_fit` (abstract implemented) or `fitMultiple` (if not overridden, [inherited]) depending on what the type of `params` is.
>         :param params: an optional param map that overrides embedded params. If a list/tuple of
>                       param maps is given, this calls fit on each param map and returns a list of
>                       models.



[KerasFitParams]:https://github.com/databricks/spark-deep-learning/blob/aeff9c9e2c24ed09c093414a7b51ec9bd42f1e86/python/sparkdl/param/shared_params.py#L150-L151
[estimator:fit]:https://github.com/apache/spark/blob/master/python/pyspark/ml/base.py#L110-L135
[inherited]:https://github.com/apache/spark/blob/30fcdc0/python/pyspark/ml/base.py#L90-L108

Reply:
Oops you're totally right, didn't realize `fit` had a params argument :P - thanks for clarifying & sorry for the confusion!